eva-server/src/main/java/uk/ac/ebi/eva/server/ws/ga4gh/beaconv2/BeaconServiceV2.java

```diff
@@ -0,0 +1,356 @@
+                                    String alternateBases, String variantType, String assemblyId,
+                                    String includeDatasetResponses) {
+        if (chromosome == null || chromosome.length() == 0) {
+            String errorMessage = "Please provide a valid referenceName type from ";
+            return errorMessage + String.join(", ", Arrays.asList(Chromosome.values()).toString());
+        } else {
+            try {
+                Chromosome.fromValue(chromosome);
+            } catch (Exception e) {
+                String errorMessage = "Please provide a valid referenceName type from ";
+                return errorMessage + String.join(", ", Arrays.asList(Chromosome.values()).toString());
+            }
+        }
+        if (assemblyId == null || assemblyId.length() == 0) {
+            return "An assemblyId must be provided";
+        }
+
+        if (referenceBases == null || referenceBases.length() == 0) {
+            return "ReferenceBases must be provided";
+        }
+
+        if (start != null && start < 0) {
+            return "Please provide a positive start number";
+        }
+
+        if (end != null && end < 0) {
+            return "Please provide a positive end number";
+        }
+
+        if (alternateBases == null && variantType == null) {
+            return "Either alternateBases or variantType is required";
+        }
+
+        if (variantType != null) {
+            try {
+                VariantType.valueOf(variantType);
+            } catch (Exception e) {
+                String errorMessage = "Please provide a valid variant type from ";
+                return errorMessage + String.join(", ", Arrays.asList(VariantType.values()).toString());
+            }
+        }
+
+        if (includeDatasetResponses != null) {
+            try {
+                BeaconAlleleRequest.IncludeDatasetResponsesEnum includeDatasetResponsesEnum =
+                        BeaconAlleleRequest.IncludeDatasetResponsesEnum.valueOf(includeDatasetResponses);
+            } catch (Exception e) {
+                String errorMessage = "Please provide a valid includeDatasetResponses from ";
+                return errorMessage + String.join(", ", Arrays.asList(BeaconAlleleRequest.
+                        IncludeDatasetResponsesEnum.values()).toString());
+            }
+        }
+        return null;
+    }
+
+    private ResponseEntity<List<BeaconAlleleResponse>> getQueryResponseEntity(Boolean exists,
+                                                                              BeaconAlleleRequest request,
+                                                                              List<BeaconDatasetAlleleResponse>
+                                                                                      datasetAlleleResponses,
+                                                                              String errorMessage) {
+        if (errorMessage != null) {
```

Comment:
We should expect the API to return with error the least amount of times, so could you please write the == branch first?

Reply:
Fixed :+1:

Reply:
fixed :)
I have swapped the order checking error message in the `queryGet` function as well